### PR TITLE
fix: refactor validating webhook to check for unique registry names, fixes RHOAIENG-13171

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description

Refactor validating webhook to check for unique registry names
Fixes RHOAIENG-13171
WIP

## How Has This Been Tested?

Tested locally, working on adding unit test for refactored webhook. 

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dhirajsb** [comment]
Verified that validating webhook works locally. Testing in another namespace `my-dsp-project`:
```shell
$ kc apply -k config/samples/secure-db/mysql-tls/
configmap/modelregistry-sample-environment created
secret/model-registry-db configured
service/model-registry-db configured
persistentvolumeclaim/model-registry-db unchanged
deployment.apps/model-registry-db configured
The ModelRegistry "modelregistry-sample" is invalid: metadata.name: Duplicate value: "registry named modelregistry-sample already exists in namespace odh-model-registries"
```

File: api/v1alpha1/modelregistry_webhook.go [hunk 1]
```diff
@@ -243,3 +219,81 @@ func (r *ModelRegistry) ValidateIstioConfig() (warnings admission.Warnings, err
...
+	r, ok := obj.(*ModelRegistry)
+	if !ok {
+		return nil, errors.NewBadRequest(fmt.Sprintf("unsupported type %s", obj.GetObjectKind().GroupVersionKind().String()))
+	}
+	return r, nil
+}
+
+func (v *modelRegistryValidator) validateName(ctx context.Context, r *ModelRegistry) (warnings admission.Warnings, err error) {
+	registries := &ModelRegistryList{}
+	err = v.Client.List(ctx, registries)
```
[the remark this answers is not in the document]
Comment (dhirajsb) [reply api/v1alpha1/modelregistry_webhook.go:260]: Yes, operator's RBAC allows it to list all MRs.

**Comment by dhirajsb** [comment]
@tarilabs yes, gateway routes are created with the model registry name as the host name. We could add the namespace suffix to it to make it unique, but it would make the host names tightly coupled to the namespace where it's deployed and leaking a deployment detail. 
Also, model registries in our multi-tenancy model are cluster wide resources (i.e. not only meant for the namespace where they are deployed), hence the unique name requirement.

**Comment by dhirajsb** [comment]
@Al-Pragliola did you get a chance to review this PR? I'd like to merge this and #148 today.

**Review by Al-Pragliola — APPROVED** [review]
lgtm, kudos for the added test

**Comment by dhirajsb** [comment]
Yeah, we need to start adding more tests there, hence the TODO reminder :smile

**Pull request merged**